File: src/Controls/KarveControls/DataGrid/DataGridHelpers/DataGridColumnVirtual/DataGridColumnVirtuales.cs
using System.Collections;
using System.Collections.Generic;
using System.Linq;

namespace KarveControls.DataGrid.DataGridHelpers.DataGridColumnVirtual
{
    public class DataGridColumnVirtuales : Dictionary<string, DataGridColumnVirtual>
    {

        public void Add(DataGridColumnVirtual TablaQuery)
        {
            base.Add(TablaQuery.Name, TablaQuery);
        }

        public ArrayList Order()
        {
            dynamic HC = (from c in Values orderby c.Item select c);
            ArrayList RS = new ArrayList();
            foreach (DataGridColumnVirtual ctr in HC)
            {
                RS.Add(ctr);
            }
            return RS;
        }
    }
}

File: documentation/Karve/CustomControls/DataGrid/Clases/DataGridColumnVirtual/DataGridColumnVirtuales.vb
Public Class DataGridColumnVirtuales
    Inherits System.Collections.Hashtable

    Public Overloads Sub Add(ByVal TablaQuery As DataGridColumnVirtual)
        MyBase.Add(TablaQuery.Name, TablaQuery)
    End Sub

    Public Function Order() As ArrayList
        Dim HC = (From c As DataGridColumnVirtual In Me.Values Order By c.Item Select c)
        Dim RS As New ArrayList
        For Each ctr As DataGridColumnVirtual In HC
            RS.Add(ctr)
        Next
        Return RS
    End Function

End Class
